mxcache-generation/src/test/java/com/maxifier/mxcache/template/impl/PInlineDependencyCache.template

```diff
@@ -68,6 +76,26 @@ public class #E#InlineDependencyCache extends #E#InlineCacheImpl implements Depe
             dependentNodes = new THashSet<Reference<DependencyNode>>();
         }
         dependentNodes.add(node.getSelfReference());
+        cleanupIfNeeded();
+    }
+
+    private void cleanupIfNeeded() {

```

Comment:
There are two different concepts:
- `Cache` - takes care of storing the data and accessing it;
- `DependencyNode` - takes care about dependencies.

There are few implementations `XXXInlineDependencyCache` that implement both `Cache` and `DependencyNode`. This reduces the memory overhead of caching in some simple cases.

But `cleanupIfNeeded` stuff is not part of `Cache` contract, so `AbstractCache` has nothing to do with that.